controller/appcontroller.go

```diff
@@ -916,6 +916,10 @@ func (ctrl *ApplicationController) refreshAppConditions(app *appv1.Application)
 			})
 		}
 	} else {
+		destCondition := argo.ValidateDestination(context.Background(), &app.Spec.Destination, ctrl.db)
```

Comment:
We need to infer server field in `processAppOperationQueueItem` method too. E.g. finalizeApplicationDeletion uses `Destination.Server` field.